engine/graphics/proto/graphics/graphics_ddf.proto

```diff
@@ -353,22 +353,25 @@ message ShaderDesc
 
     message ResourceBinding
     {
-        required string         name          = 1;
-        required uint64         name_hash     = 2;
-        required ResourceType   type          = 3;
-        optional uint32         set           = 4;
-        optional uint32         binding       = 5;
+        required string         name               = 1;
+        required uint64         name_hash          = 2;
+        required ResourceType   type               = 3;
+        optional uint32         set                = 4;
+        optional uint32         binding            = 5;
+        optional uint32         element_count      = 6;
+        optional string         instance_name      = 7;
+        optional uint64         instance_name_hash = 8;
```

Comment:
These are set to Optional, it's only exposed in this scenario:

uniform Lights {
....
} scene_lights[4];